extensions/sql-database-projects/src/models/dataSources/sqlConnectionStringSource.ts

```diff
@@ -25,6 +25,27 @@ export class SqlConnectionDataSource extends DataSource {
 		return constants.sqlConnectionStringFriendly;
 	}
 
+	public get server(): string {
+		return this.getSetting(constants.dataSourceSetting);
+	}
+
+	public get database(): string {
+		return this.getSetting(constants.initialCatalogSetting);
+	}
+
+	public get integratedSecurity(): boolean {
+		return this.getSetting(constants.integratedSecuritySetting).toLowerCase() === 'true';
+	}
+
+	public get username(): string {
+		return this.getSetting(constants.userIdSetting);
+	}
+
+	public get password(): string {
+		// TODO: secure password storage
+		return this.getSetting(constants.passwordSetting);
```

Comment:
azdata already has APIs to store, load password in secure storage

Reply:
add a task for this also if you cant take in the PR?

Reply:
Created #10561